fastly/tls_subscription.go

```diff
@@ -0,0 +1,224 @@
+		"include":                s.Include,
+		"page[number]":           s.PageNumber,
+		"page[size]":             s.PageSize,
+		"sort":                   s.Sort,
+	}
+
+	for key, v := range pairings {
+		switch value := v.(type) {
+		case string:
+			if value != "" {
+				result[key] = value
+			}
+		case int:
+			if value != 0 {
+				result[key] = strconv.Itoa(value)
+			}
+		}
+	}
+	return result
+}
+
+// ListTLSSubscriptions lists all managed TLS subscriptions
+func (c *Client) ListTLSSubscriptions(i *ListTLSSubscriptionsInput) ([]*TLSSubscription, error) {
+	response, err := c.Get("/tls/subscriptions", &RequestOptions{
+		Params: i.formatFilters(),
+		Headers: map[string]string{
+			"Accept": "application/vnd.api+json", // Needed for "include" but seemingly not the other fields
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := jsonapi.UnmarshalManyPayload(response.Body, reflect.TypeOf(new(TLSSubscription)))
+	if err != nil {
+		return nil, err
+	}
+
+	// Convert slice of interface{}s to a slice of TLSSubscription structs
+	subscriptions := make([]*TLSSubscription, len(data))
+	for i := range data {
+		typed, ok := data[i].(*TLSSubscription)
+		if !ok {
+			return nil, fmt.Errorf("unexpected response type: %T", data[i])
+		}
+		subscriptions[i] = typed
+	}
+
+	return subscriptions, nil
+}
+
+type CreateTLSSubscriptionInput struct {
+	// ID value is ignored and should not be set, needed to make JSONAPI work correctly.
+	ID string `jsonapi:"primary,tls_subscription"`
+	// CertificateAuthority is the entity that issues and certifies the TLS certificates for your subscription. Valid values are lets-encrypt or globalsign.
+	CertificateAuthority string `jsonapi:"attr,certificate_authority,omitempty"`
+	// Configuration options that apply to the enabled domains on this subscription. Only ID needs to be populated
+	Configuration *TLSConfiguration `jsonapi:"relation,tls_configuration,omitempty"`
+	// CommonName is the common name associated with the subscription generated by Fastly TLS. Must be included in Domains. Only the ID fields of each one need to be set.
+	CommonName *TLSDomain `jsonapi:"relation,common_name,omitempty"`
+	// Domains list to enable TLS for. Only the ID fields of each one need to be set.
+	Domains []*TLSDomain `jsonapi:"relation,tls_domain"`
+}
+
+func (c *Client) CreateTLSSubscription(i *CreateTLSSubscriptionInput) (*TLSSubscription, error) {
+	if len(i.Domains) == 0 {
+		return nil, ErrMissingTLSDomain
+	}
+	if i.CommonName != nil && !domainInSlice(i.Domains, i.CommonName) {
+		return nil, ErrCommonNameNotInDomains
+	}
+
+	response, err := c.PostJSONAPI("/tls/subscriptions", i, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var subscription TLSSubscription
+	err = jsonapi.UnmarshalPayload(response.Body, &subscription)
+	if err != nil {
+		return nil, err
+	}
+
+	return &subscription, nil
+}
+
+func domainInSlice(haystack []*TLSDomain, needle *TLSDomain) bool {
```

Comment:
Could we add in the missing description please. I appreciate it's not publicly exported, but as mentioned earlier ...I'm a stickler for the docs 🙂